main/lsp/requests/document_symbol.cc

```diff
@@ -77,6 +79,27 @@ bool DocumentSymbolTask::isDelayable() const {
     return true;
 }
 
+bool isOwnerInTheSameFile(const core::GlobalState &gs, core::SymbolRef ref, core::FileRef fref) {
+    auto locs = ref.data(gs)->owner.data(gs)->locs();
```

Comment:
I think this might be copying the vector? If you take it by reference it will definitely not copy the vector:

```suggestion
    auto &locs = ref.data(gs)->owner.data(gs)->locs();
```